app/src/main/java/nl/mpcjanssen/simpletask/Simpletask.kt

```diff
@@ -1472,23 +1443,21 @@ class Simpletask : ThemedNoActionBarActivity() {
         val ellipsizeKey = m_app.getString(R.string.task_text_ellipsizing_pref_key)
         val ellipsizePref = Config.prefs.getString(ellipsizeKey, noEllipsizeValue)
 
-        if (noEllipsizeValue != ellipsizePref) {
+        if (noEllipsizeValue != ellipsizePref) elipsis@ {
             val truncateAt: TextUtils.TruncateAt?
-            when (ellipsizePref) {
-                "start" -> truncateAt = TextUtils.TruncateAt.START
-                "end" -> truncateAt = TextUtils.TruncateAt.END
-                "middle" -> truncateAt = TextUtils.TruncateAt.MIDDLE
-                "marquee" -> truncateAt = TextUtils.TruncateAt.MARQUEE
-                else -> truncateAt = null
+            taskText.ellipsize = when (ellipsizePref) {
+                "start" -> TextUtils.TruncateAt.START
+                "end" -> TextUtils.TruncateAt.END
+                "middle" -> TextUtils.TruncateAt.MIDDLE
+                "marquee" -> TextUtils.TruncateAt.MARQUEE
+                else -> {
+                    log.warn(TAG, "Unrecognized preference value for task text ellipsis: {} ! $ellipsizePref")
+                    return@elipsis
+                }
             }
 
-            if (truncateAt != null) {
-                taskText.maxLines = 1
-                taskText.setHorizontallyScrolling(true)
-                taskText.ellipsize = truncateAt
-            } else {
-                log.warn(TAG, "Unrecognized preference value for task text ellipsis: {} !" + ellipsizePref)
-            }
+            taskText.maxLines = 1
+            taskText.setHorizontallyScrolling(true)
```

Comment:
This is also some style; I'm willing to revert, although I don't love how it was originally, either.